instrumentation.go

```diff
@@ -396,3 +408,13 @@ func WithSampler(sampler trace.Sampler) InstrumentationOption {
 		return c, nil
 	})
 }
+
+// WithResourceAttributes returns an [InstrumentationOption] that will configure
+// an [Instrumentation] to use the provided attributes as OpenTelemetry resource
+// attributes.
```

Comment:
Can we document that the attributes are appended. Maybe:

```suggestion
// WithResourceAttributes returns an [InstrumentationOption] that will configure
// an [Instrumentation] to add the provided attributes to the OpenTelemetry resource.
```